src/main/groovy/com/morpheusdata/nutanix/prismelement/plugin/NutanixPrismElementPluginProvisionProvider.groovy

```diff
@@ -259,47 +282,430 @@ class NutanixPrismElementPluginProvisionProvider extends AbstractProvisionProvid
+				} else {
+					log.error("Error in creating the image: ${imageResults.msg}")
+				}
+			}
+		} finally {
+			if (lockId) {
+				context.releaseLock(lockKey, [lock: lockId]).blockingGet()
+			}
+		}
+		return imageExternalId
+	}
+
+	private Map buildCreateVmOpts(Cloud cloud, ComputeServer server, Workload workload, Map runWorkloadOpts) {
+		def rootVolume = workload.server.volumes?.find { it.rootVolume == true }
+		def maxStorage = rootVolume.maxStorage ?: workload.maxStorage ?: workload.instance.plan.maxStorage
+		def datastore = getDatastoreOption(cloud, server.account, rootVolume?.datastore, rootVolume?.datastoreOption, maxStorage)
+		def datastoreId = datastore?.externalId
+		def dataDisks = workload.server?.volumes?.findAll { it.rootVolume == false }?.sort { it.id }
+		def maxMemory = workload.maxMemory ?: workload.instance.plan.maxMemory
+		def coresPerSocket = workload.coresPerSocket ?: workload.instance.plan.coresPerSocket ?: 1
+		def maxCores = workload.maxCores ?: workload.instance.plan.maxCores
+
+		def createOpts = [
+			account       : server.account,
+			containerId   : datastoreId,
+			coresPerSocket: coresPerSocket,
+			dataDisks     : dataDisks,
+			domainName    : server.getExternalHostname(),
+			externalId    : server.externalId,
+			hostname      : server.getExternalHostname(),
+			isSysprep     : server.sourceImage?.isSysprep,
+			maxCores      : maxCores,
+			maxMemory     : maxMemory,
+			maxStorage    : maxStorage,
+			name          : server.name,
+			networkConfig : runWorkloadOpts.networkConfig,
+			platform      : server.serverOs?.platform ?: 'linux',
+			rootVolume    : rootVolume,
+			server        : server,
+			uefi          : server.sourceImage?.uefi,
+			uuid          : server.apiKey,
+			zone          : server.cloud,
+		]
+		createOpts.fqdn = createOpts.hostname
+		if (createOpts.domainName) {
+			createOpts.fqdn += '.' + createOpts.domainName
+		}
+		createOpts
+	}
+
+	def getDatastoreOption(Cloud cloud, Account account, DatastoreIdentity datastore, String datastoreOption, Long size) {
+		def rtn = null
+		if (datastore) {
+			rtn = datastore
+		} else if (datastoreOption == 'auto' || !datastoreOption) {
+			def datastores = context.services.cloud.datastore.list(
+				new DataQuery().withFilters(
+					new DataFilter('refType', 'ComputeZone'),
+					new DataFilter('refId', cloud.id),
+					new DataFilter('type', 'generic'),
+					new DataFilter('online', true),
+					new DataFilter('active', true),
+					new DataFilter('freeSpace', '>', size),
+					new DataOrFilter(
+						new DataFilter('owner.id', account.id),
+						new DataFilter('visibility', 'public')
+					)
+				).withSort("freeSpace", DataQuery.SortOrder.desc)
+			)
+			rtn = datastores.find { ds ->
+				ds.externalId != null && ds.externalId != ""
+			}
+		}
+		return rtn
+	}
+
+	static boolean isCloudInitIso(Map createOpts) {
+		def rtn = false
+		if (createOpts.platform == 'windows' && createOpts.isSysprep != true) {
```

Comment:
It's cloudinit if it's windows and not sysprep? Why?

Reply:
From what I can tell, cloud init on windows needs to use a cloudinit ISO. So if we're not using sysprep this is the alternative.